backend/question-service/controllers/question.controller.js

```diff
@@ -0,0 +1,117 @@
+const db = require('../models');
+const Question = db.questions;
+
+exports.create = (req, res) => {
+  if (!req.body.questionId
+    || !req.body.questionTitle
+    || !req.body.questionDescription
+    || !req.body.questionCategory
+    || !req.body.questionComplexity) {
+    res.status(400).send({message: 'Question details missing.'});
+    return;
+  }
+
+  const question = new Question({
```

Comment:
This logic can be transferred to question.model.js. You can create functions in question.model.js handling DB logic while the controller simply handles incoming and outgoing HTTP requests. However, this is up to you  since the code is not too long anyway.

Reply:
Gave it a shot, but it turned out to be quite clunky and didn't quite work out, if you have a suggestion in mind though I'd be happy to use it